sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/PooledHashMapTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PooledHashMapTest {
```

Comment:
Are there any general purpose map test suites to test this implementation with more rigor?

Reply:
I thought about it for some time and couldn't up with any additional tests. It cover the functionality in the test. The "pool" effect is tested in the special purpose GC benchmark test, and also in a purpose built test `collect_reusableData_reusedObjectsAreReturnedOnSecondCall` in `AsynchronousMetricStorageTest`. Also `AsynchronousMetricStorageTest` tests are run twice, one for immutable and one for reusable.

Together it covers it all, no?